File: plugins/OrchidKSS/src/main/java/com/eden/orchid/kss/menu/StyleguideSectionsMenuItemType.java
package com.eden.orchid.kss.menu;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.theme.menus.menuItem.OrchidMenuItemImpl;
import com.eden.orchid.api.theme.menus.menuItem.OrchidMenuItem;
import com.eden.orchid.kss.KssGenerator;
import com.eden.orchid.kss.KssPage;

import javax.inject.Inject;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class StyleguideSectionsMenuItemType extends OrchidMenuItem {

    @Inject
    public StyleguideSectionsMenuItemType(OrchidContext context) {
        super(context, "styleguideSections", 100);
    }

    @Override
    public List<OrchidMenuItemImpl> getMenuItems() {
        List<OrchidMenuItemImpl> menuItems = new ArrayList<>();

        for (Map.Entry<String, List<KssPage>> section : KssGenerator.sections.entrySet()) {
            if(section.getValue().size() > 0) {
                menuItems.add(new OrchidMenuItemImpl(context, section.getValue().get(0)));
            }
        }

        return menuItems;
    }
}

File: plugins/OrchidKSS/src/main/kotlin/com/eden/orchid/kss/menu/StyleguideSectionsMenuItemType.kt
package com.eden.orchid.kss.menu

import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.options.annotations.Description
import com.eden.orchid.api.options.annotations.Option
import com.eden.orchid.api.options.annotations.StringDefault
import com.eden.orchid.api.theme.menus.MenuItem
import com.eden.orchid.api.theme.menus.OrchidMenuFactory
import com.eden.orchid.api.theme.pages.OrchidPage
import com.eden.orchid.kss.model.KssModel

@Description("Link to all the sections of your styleguide.", name = "Styleguide Sections")
class StyleguideSectionsMenuItemType : OrchidMenuFactory("styleguideSections") {

    @Option
    @StringDefault("Styleguide")
    @Description("The title of the root menu item.")
    lateinit var title: String

    override fun getMenuItems(
        context: OrchidContext,
        page: OrchidPage
    ): List<MenuItem> {
        val model = context.resolve(KssModel::class.java)

        val menuItems = ArrayList<MenuItem>()

        model.sections.keys.forEach {
            val pages = model.getSectionRoot(it)
            if (pages.isNotEmpty()) {
                menuItems.add(
                    MenuItem.Builder(context)
                        .title(title)
                        .pages(pages)
                        .build()
                )
            }
        }

        return menuItems
    }
}

